packages/medusa/src/repositories/customer-group.ts

```diff
@@ -1,5 +1,64 @@
-import { EntityRepository, Repository } from "typeorm"
+import {
+  EntityRepository,
+  Repository,
+  getManager,
+  getConnection,
+} from "typeorm"
+import { MedusaError } from "medusa-core-utils"
 import { CustomerGroup } from "../models/customer-group"
+import { CustomerGroupsBatchCustomer } from "../types/customer-groups"
+import { Customer } from ".."
 
 @EntityRepository(CustomerGroup)
-export class CustomerGroupRepository extends Repository<CustomerGroup> {}
+export class CustomerGroupRepository extends Repository<CustomerGroup> {
+  async addCustomerBatch(
+    groupId: string,
+    customerIds: CustomerGroupsBatchCustomer[]
+  ): Promise<CustomerGroup> {
+    try {
+      const customerGroup = await this.findOne(groupId)
+      if (!customerGroup) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `CustomerGroup with id ${groupId} was not found`
+        )
+      }
+
+      await this.createQueryBuilder()
+        .insert()
+        .into("customer_group_customers")
+        .values(
+          customerIds.map(({ id }) => ({
+            customer_id: id,
+            customer_group_id: groupId,
+          }))
+        )
+        .orIgnore()
+        .execute()
+
+      return customerGroup
+    } catch (error) {
+      if (error.code === "23503") {
+        const existingCustomers = await getConnection()
+          .getRepository(Customer)
+          .createQueryBuilder("customer")
```

Comment:
We need to ensure that this works within a transaction - I am not entirely sure if `getConnection` is scoped to do so.

I think it might work to just use `this.createQueryBuilder().from(Customer)` - I have done a similar thing here: https://github.com/medusajs/medusa/blob/tax-api/packages/medusa/src/repositories/tax-rate.ts

Reply:
Did you find out if we can avoid `getConnection`?